packages/react/src/useEditor.ts

```diff
@@ -113,14 +114,130 @@ export const useEditor = (options: Partial<EditorOptions> = {}, deps: Dependency
 
     return () => {
       isMounted = false
+      editorRef.current?.destroy()
     }
   }, deps)
 
+  return editorRef.current
+}
+
+/**
+ * This hook will create and immediately return a defined editor instance. In future major versions, this
+ * hook will be removed and useEditor will be changed to behave like this hook.
+ */
+export const useEditorForImmediateRender = (options: Partial<EditorOptions> = {}, deps: DependencyList = []) => {
+  const {
+    onBeforeCreate,
+    onBlur,
+    onCreate,
+    onDestroy,
+    onFocus,
+    onSelectionUpdate,
+    onTransaction,
+    onUpdate,
+  } = options
+
+  const onBeforeCreateRef = useRef(onBeforeCreate)
+  const onBlurRef = useRef(onBlur)
+  const onCreateRef = useRef(onCreate)
+  const onDestroyRef = useRef(onDestroy)
+  const onFocusRef = useRef(onFocus)
+  const onSelectionUpdateRef = useRef(onSelectionUpdate)
+  const onTransactionRef = useRef(onTransaction)
+  const onUpdateRef = useRef(onUpdate)
+
+  const isMounted = useRef(false)
+
   useEffect(() => {
+    isMounted.current = true
+
     return () => {
-      return editorRef.current?.destroy()
+      isMounted.current = false
     }
   }, [])
 
-  return editorRef.current
+  const [, forceUpdate] = useState({})
+  const editor = useMemo(() => {
+    const instance = new Editor(options)
+
+    instance.on('transaction', () => {
+      requestAnimationFrame(() => {
+        requestAnimationFrame(() => {
+          if (isMounted.current) {
+            forceUpdate({})
```

Comment:
Not required in this PR, but it would be nice to add comment here why is this section necessary (to call the forceUpdate wrapped in 2 requestAnimationFrame calls on every editor transaction). This is a bit black magic to me.

Reply:
Same here. I just copied this over from the original hook, but it looks to me like it's logic to get around timing issues between Prosemirror transactions and React render cycles

Reply:
This seems to solve the flickering issue mentioned here: https://github.com/ueberdosis/tiptap/issues/2040

Reply:
Unfortunately there is a habit in this repo of closing non-fixed issues.

@szymonchudy is that issue separate from #5166? It's hard to tell if they're related or truly the same.